src/ui/src/builder/useComponentFieldViewModel.spec.ts

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { describe, it, expect, vi } from "vitest";
+import { ref, reactive, nextTick } from "vue";
+import { useComponentFieldViewModel } from "./useComponentFieldViewModel";
+
+type Component = { id: string; type: string; content: Record<string, string> };
+type Defs = Record<string, { fields?: Record<string, { default?: string }> }>;
+
+function mockDependencies(
+	initial: Record<string, Component>,
+	defs: Defs = {},
+): { components: Record<string, Component>; wf: any; ssbm: any } {
+	const components = reactive(initial);
+
+	function getComponentById(id: string) {
+		return components[id];
+	}
+
+	function getComponentDefinition(type?: string) {
+		return type ? defs[type] : undefined;
+	}
+
+	return {
+		components,
+		wf: {
+			getComponentById,
+			getComponentDefinition,
+			sendComponentUpdate: vi.fn(),
+		},
+		ssbm: {
+			openMutationTransaction: vi.fn(),
+			registerPreMutation: vi.fn(),
+			registerPostMutation: vi.fn(),
+			closeMutationTransaction: vi.fn(),
+		},
+	};
+}
```

Comment:
We already have `buildMockCore` that takes care of mocking API call

```suggestion
function mockDependencies(
	initial: Record<string, Component>,
): { components: Record<string, Component>; wf: any; ssbm: any } {
	const components = reactive(initial);
	
	const mockCore = buildMockCore();
	for (const comp of Object.values(initial)) mockCore.core.addComponent(comp);

	return {
		components,
		wf: mockCore.core,
		ssbm: {
			openMutationTransaction: vi.fn(),
			registerPreMutation: vi.fn(),
			registerPostMutation: vi.fn(),
			closeMutationTransaction: vi.fn(),
		},
	};
}
```

In the same vein, it could be the right time to introduce `buildMockBuilderManager` that can be reused.

Reply:
there is `generateBuilderManager` is already in place